scripts/check_compliance.py

```diff
@@ -165,6 +168,16 @@ def run(self):
         # single Kconfig tree and warns for all references to undefined symbols
         os.environ["KCONFIG_STRICT"] = "y"
 
+        kconfig_env_var_list_path = os.path.join(self.zephyr_base, "cmake/kconfig_env_vars.txt")
+        kconfig_env_vars = set(open(kconfig_env_var_list_path).read().splitlines())
```

Comment:
The file should be closed, is it? I can't tell.

Reply:
yup it should probably do a close call as I'm not using `with` I'm too used to using `with` keyword which does this for you. Not sure where the close should be put I guess after `read`

Reply:
In CPython (the default Python implementation), files will be closed immediately when the last reference to them goes away (which is after that line).

The reason it's still a good idea (but not always super important) to use `with open(...) ...` is that some other Python implementations don't use simple reference counting, like PyPy.

Reply:
`with` is used like this by the way:

```python
with open(...) as f:
    f.read(), etc.
```